neuraxle/checkpoints.py

```diff
@@ -96,44 +102,45 @@ def __init__(self, force_checkpoint_name: str = None, cache_folder: str = DEFAUL
         self.cache_folder = cache_folder
         self.force_checkpoint_name = force_checkpoint_name
 
-    def read_checkpoint(self, data_inputs):
+    def read_checkpoint(self, data_container: DataContainer):
         """
         Read pickle files for data inputs and expected outputs checkpoint
         :return: tuple(data_inputs, expected_outputs
         """
-        data_inputs_checkpoint_file_name = self.checkpoint_path
-        with open(self.get_checkpoint_file_path(data_inputs_checkpoint_file_name), 'rb') as file:
-            data_inputs = pickle.load(file)
+        with open(self.get_checkpoint_file_path(data_container), 'rb') as file:
+            checkpoint = pickle.load(file)
 
-        return data_inputs
+        return checkpoint
 
-    def save_checkpoint(self, data_inputs):
+    def save_checkpoint(self, data_container: DataContainer):
         """
         Save pickle files for data inputs and expected output
         to create a checkpoint
-        :param data_inputs: data inputs to be saved in a pickle file
+        :param data_container: data inputs to be saved in a pickle file
         :return:
         """
-        # TODO: don't force the user to set the checkpoint name (use step name instead).
         self.set_checkpoint_path(self.force_checkpoint_name)
-        with open(self.get_checkpoint_file_path(data_inputs), 'wb') as file:
-            pickle.dump(data_inputs, file)
+        with open(self.get_checkpoint_file_path(data_container), 'wb') as file:
+            pickle.dump(data_container, file)
 
     def set_checkpoint_path(self, path):
         """
         Set checkpoint path inside the cache folder (ex: cache_folder/pipeline_name/force_checkpoint_name/data_inputs.pickle)
         :param path: checkpoint path
         """
+        if path is None:
```

Comment:
clean code : use self.force_checkpoint_name directly for clarity

Reply:
TODO: init checkpoint path at the `initialize` method?